lrlex/src/lib/builder.rs

```diff
@@ -169,12 +174,14 @@ where
         // Token IDs
         if let Some(ref rim) = self.rule_ids_map {
             for (n, id) in rim {
-                outs.push_str(&format!(
-                    "#[allow(dead_code)]\npub const T_{}: {} = {:?};\n",
-                    n.to_ascii_uppercase(),
-                    StorageT::type_name(),
-                    *id
-                ));
+                if RE_TOKEN_ID.is_match(n) {
```

Comment:
Will this just silently ignore my tokens if I used weird names? In that case we might want to throw an error message or something so people are not confused if something doesn't work as expected.

Reply:
It does silently ignore them, but I'm not really sure what else to sensibly do: unless you're using generic parse trees, you pretty much don't care about these. And if you use generic parse trees, you'll very quickly realise why there isn't a variable called `T=`.

Reply:
Can you remind me again what these `T_` variables are used for? So it won't happen that the program compiles but behaves weirdly because we ignore variables with weird names?

Reply:
They tell you the `TIdx` of a token name. If you have a generic parse tree and want to (say) check if a terminal is an ID, you need to use these. Otherwise you can ignore them.

Reply:
Ah I see. In that case I agree that there is probably no need to raise an error. I will merge as is then.